### DISCUSSION TITLE
Per-user File Manager Permissions

### DISCUSSION THREAD

**Discussion by YelehaUwU** [pelican-dev/panel#616] [Ideas]
It would be neat to have a button for each user, that would open a menu, where one could pick specific files&folders that specific user will not have access to per their permissions, and will not even see in the file manager.
### Examples

**All blocked**
```
*
```
**All blocked except the contents of folder alpha1**
```
*
!/path/to/alpha1
```
**All blocked except the contents of all file extensions of type .cpp**
```
*
!*.cpp
```
_(Or all of this could be in a pretty interface instead of typing paths, like opening the file manager and selecting specific folders there with a checkbox.)_

**Comment by QuintenQVD0** [comment]
You can do that at egg level already not yet on user level.

**Reply by YelehaUwU** [reply]
It'd be fit for server owners, so that they can block people from seeing certain credentials, or modifying core components, as many games and other eggs don't have the ability to read environment variables for tokens and credentials.

**Reply by QuintenQVD0** [reply]
That sounds more gloval then uset specific

**Reply by YelehaUwU** [reply]
Specific people should have different permissions in the server, this was already discussed in Pterodactyl so I'm just speaking on behalf of [the old post](https://github.com/pterodactyl/panel/issues/2757) and personal experience.

**Comment by notAreYouScared** [comment]
If you can't trust them to see sensitive files, you shouldn't trust them with file access.... We have no intention of adding this at this time.

**Reply by YelehaUwU** [reply]
Many hosting providers give you this option to allow limiting access, it's basically become a golden standard.

**Comment by YelehaUwU** [comment]
https://github.com/pterodactyl/panel/issues/2757

Its still getting attention on pterodactyl, perhaps it should be re-considered.
